src/components/Page/components/Header/Header.tsx

```diff
@@ -228,6 +214,42 @@ export function Header({
   );
 }
 
+function PrimaryActionMarkup({
+  primaryAction,
+}: {
+  primaryAction: PrimaryAction | React.ReactNode;
+}) {
+  const {isNavigationCollapsed} = useMediaQuery();
+  const {newDesignLanguage} = useFeatures();
+  let content = primaryAction;
+  if (isPrimaryAction(primaryAction)) {
```

Comment:
Nitpick: With an early return we don't have to wrap the entire component 😄 

```suggestion
  let content = primaryAction;
  if (!isPrimaryAction(primaryAction)) return null
  
  const primary = ...
```

Reply:
I never want to return `null` with this component. This is wrapped in the condition because it changes `content` to be the button instead of `primaryAction` (which could be a React node). The `ConditionalWrapper` should always be returned with whatever `content` is as the children.